src/Routing/Route/Route.php

```diff
@@ -705,11 +707,14 @@ public function match(array $url, array $context = []): ?string
             return null;
         }
 
-        // check patterns for routed params
+        // check patterns for routed params, if they are not the default value
         if (!empty($this->options)) {
             foreach ($this->options as $key => $pattern) {
-                if (isset($url[$key]) && !preg_match('#^' . $pattern . '$#u', (string)$url[$key])) {
-                    return null;
+                if (
```

Comment:
The conditions for this `if` are practically unreadable. It would be better to split the checks and assign them to well named variables and then use them inside the `if` so that it's easier to understand what's going on here.